### DISCUSSION TITLE
Sonar-Gerrit integration in Jenkins

### DISCUSSION THREAD

**Discussion by anjithachinnu** [jenkinsci/sonar-gerrit-plugin#134] [Q&A]
Hi,

There is a Jenkins freestyle job that is triggered by some Gerrit events.In response to triggers, it downloads the changes and runs a Sonarqube Analysis. The Sonarqube comments detailing the results are to be posted on Gerrit after the analysis is complete.This is the expected behaviour.

As of now, In Jenkins, the Sonarqube execution is success. But the build is failed mentioning the error 
"
```
*ERROR: Build step failed with exception
java.lang.IllegalArgumentException: Gerrit change number is empty*
```

Could anyone please suggest on this?

**Comment by ranyan713** [comment]
Did you pass this property?
```
Key	Recommended value template	Example
sonar.pullrequest.key	${GERRIT_CHANGE_NUMBER}-${GERRIT_PATCHSET_NUMBER}	250-1
```
